docs/kubernetes/README.md

```diff
@@ -123,6 +123,38 @@ parameters:
 
 Your nodes might need to have `cryptsetup` installed to mount the volumes with LUKS.
 
+### XFS Filesystem
+
+XFS can be enabled by creating a custom storage class and settting the `csi.storage.k8s.io/fstype` parameter to `xfs`.
+
+If all nodes in your cluster run the same kernel version, the CSI driver will automatically determine the correct `mkfs.xfs` options. 
```

Comment:
Should we put this feature behind a flag? The magic of this feature might be too much for some users.